src/Sulu/Component/Content/Types/ResourceLocator.php

```diff
@@ -11,213 +11,27 @@
-        } catch (ResourceLocatorNotFoundException $ex) {
-            $value = null;
-        }
-
-        return $value;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function loadHistoryByUuid($uuid, $webspaceKey, $languageCode, $segmentKey = null)
-    {
-        return $this->getStrategy()->loadHistoryByContentUuid($uuid, $webspaceKey, $languageCode, $segmentKey);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function deleteByPath($path, $webspaceKey, $languageCode, $segmentKey = null)
-    {
-        $this->getStrategy()->deleteByPath($path, $webspaceKey, $languageCode, $segmentKey);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function restoreByPath($path, $webspaceKey, $languageCode, $segmentKey = null)
-    {
-        $this->getStrategy()->restoreByPath($path, $webspaceKey, $languageCode, $segmentKey);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function write(
-        NodeInterface $node,
-        PropertyInterface $property,
-        $userId,
-        $webspaceKey,
-        $languageCode,
-        $segmentKey = null
-    ) {
-        $value = $property->getValue();
-
-        if ($value === null || $value === '') {
-            $this->remove($node, $property, $webspaceKey, $languageCode, $segmentKey);
-
-            return;
-        }
-
-        $treeValue = $this->getResourceLocator($node, $webspaceKey, $languageCode, $segmentKey);
-        if ($treeValue === '/') {
-            return;
-        }
-
-        // only if property value is the same as tree value (is different in move / copy / rename workflow)
-        // or the tree value does not exist
-        if ($treeValue === null) {
-            $this->getStrategy()->save($node, $value, $userId, $webspaceKey, $languageCode, $segmentKey);
-        } elseif ($treeValue !== $value) {
-            $this->getStrategy()->move(
-                $treeValue,
-                $value,
-                $node,
-                $userId,
-                $webspaceKey,
-                $languageCode,
-                $segmentKey
-            );
-        }
-
-        $node->setProperty($property->getName(), $value);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function remove(
-        NodeInterface $node,
-        PropertyInterface $property,
-        $webspaceKey,
-        $languageCode,
-        $segmentKey = null
-    ) {
-        if ($node->hasProperty($property->getName())) {
-            $node->getProperty($property->getName())->remove();
-        }
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function loadContentNodeUuid($resourceLocator, $webspaceKey, $languageCode, $segmentKey = null)
-    {
-        return $this->getStrategy()->loadByResourceLocator($resourceLocator, $webspaceKey, $languageCode, $segmentKey);
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getStrategy()
-    {
-        // TODO get strategy from ???
```

Comment:
I'd say we do that via the container :-)

If we have more than one strategy one day we can still introduce an alias for that.